### PR TITLE
[Feature] refractor dist came; fix percision error; add low level zero test with bert model zoo;

### PR DESCRIPTION
**Pull request by duanjunwen**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs
- [ ] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/nn/optimizer/came.py [hunk 1]
```diff
@@ -120,7 +121,7 @@ def step(self, closure=None):
                     exp_avg_sq.mul_(group["betas"][1]).add_(update, alpha=1.0 - group["betas"][1])
                     update = exp_avg_sq.rsqrt().mul_(grad)
 
-                update.div_((self._rms(update) / group["clip_threshold"]).clamp_(min=1.0))
+                # update.div_((self._rms(update) / group["clip_threshold"]).clamp_(min=1.0))
```
Comment (ver217) [inline colossalai/nn/optimizer/came.py:124]: Why

File: tests/test_optimizer/test_dist_came.py [hunk 2]
```diff
@@ -1,308 +1,573 @@
...
+        )
     else:
-        optim.setup_distributed(tp_group, dp_group, None, zero_flag=False)
+        shard_to_param = set_master_param_to_shard_param(tp_param_group)
+        dist_optim.setup_distributed(
+            tensor_parallel_group=tp_group,
+            data_parallel_group=dp_group,
+            shard_to_param=shard_to_param,
+            use_zero=use_zero,
+        )
```
Comment (ver217) [inline tests/test_optimizer/test_dist_came.py:321]: Low level api should not be called when using booster

File: tests/test_optimizer/test_dist_came.py [hunk 3]
```diff
@@ -1,308 +1,573 @@
...
 @rerun_if_address_is_in_use()
-def test_dist_lamb():
-    spawn(check_dist_lamb, nprocs=4)
+def test_dist_adafactor():
+    spawn(run_dist, nprocs=4)
 
 
 if __name__ == "__main__":
-    test_dist_lamb()
+    test_dist_adafactor()
```
Comment (ver217) [inline tests/test_optimizer/test_dist_came.py:573]: Wrong name?

File: tests/test_optimizer/test_dist_came.py (hunk 2 above)
Comment (duanjunwen) [reply tests/test_optimizer/test_dist_came.py]: > Low level api should not be called when using booster

Fix in https://github.com/hpcaitech/ColossalAI/pull/5676/commits/74d4facc625e236d232b3ae845431537c9e08748；

File: tests/test_optimizer/test_dist_came.py (hunk 3 above)
Comment (duanjunwen) [reply tests/test_optimizer/test_dist_came.py]: > Wrong name?

fix in https://github.com/hpcaitech/ColossalAI/pull/5676/commits/74d4facc625e236d232b3ae845431537c9e08748；

File: colossalai/nn/optimizer/came.py (hunk 1 above)
Comment (duanjunwen) [reply colossalai/nn/optimizer/came.py]: > Why

open rms for both came and dist came in https://github.com/hpcaitech/ColossalAI/pull/5676/commits/74d4facc625e236d232b3ae845431537c9e08748；

File: tests/test_optimizer/test_dist_came.py [hunk 4]
```diff
@@ -1,308 +1,564 @@
...
+        correctness_verify(p.data, tp_p.data, dtype)
+    clear_layout_converter()
+    Randomizer.reset_index()
+    torch.cuda.empty_cache()
+    print(f"Fwd/Bwd Test Pass")
+
+
+@parameterize("dtype", [torch.float16])
+@parameterize("tp_zero_size", [(1, 4)])
+def exam_dist_came_lowlevelzeroplugin(dtype: torch.dtype, tp_zero_size: tuple[int, int]):
```
Comment (ver217) [inline tests/test_optimizer/test_dist_came.py:268]: I notice you already check it with zero plugin at line 365. Can we remove this functions?

Comment (duanjunwen) [reply tests/test_optimizer/test_dist_came.py]: > I notice you already check it with zero plugin at line 365. Can we remove this functions?

Removed in https://github.com/hpcaitech/ColossalAI/pull/5676/commits/e82e2b24a87f8fbae791ebd1cb4e04369bb8b87c;

**Review by ver217 — APPROVED** [review]

**Pull request merged**